driver-core/src/main/com/mongodb/ClientSessionOptions.java

```diff
@@ -168,7 +209,13 @@ private Builder() {
     }
 
     private ClientSessionOptions(final Builder builder) {
-        this.causallyConsistent = builder.causallyConsistent;
+        if (builder.causallyConsistent != null && builder.causallyConsistent && builder.snapshot != null && builder.snapshot) {
+            throw new IllegalArgumentException("TODO"); // TODO
```

Comment:
What should the error message be and when is this TODO planned?

Reply:
Sorry, just missed this.  Added error messages are removed TODO.